dts/bindings/i3c/snps,designware-i3c.yaml

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2020 Samsung Electronics Co., Ltd.
+# Copyright (C) 2023 Meta Platforms
+# SPDX-License-Identifier: Apache-2.0
+
+description: >
+    This is a binding for the Synopsys I3C interface
+
+compatible: "snps,designware-i3c"
+
+include: i3c-controller.yaml
+
+properties:
+  input-clock-frequency:
+    required: true
+    type: int
+    description: clk frequency.
```

Comment:
this is a misuse of DT, we should just take a `clocks` property and obtain its rate.

Reply:
I really wish that was possible, but using the clock api is rather vendor specific and how the clock is obtain would be implementation defined for this IP

Reply:
I agree, clock control API is not _ready_ for mainstream use. I wonder if we should introduce a wrapper pattern like in CAN MCAN, where clock info is part of the wrapper ops. Note that for Nordic we'll need some minimal wrapper code (IP enablement, and IRQ clearing procedure).

Reply:
I took a look at the CAN MCAN driver... but I don't see what you are referring to with the wrapper pattern..

As for obtaining the input `sys_clk` rate to the ip... I don't see the current clock_control api working from a 'vendor-agnostic' ip such as synopsys designment.... but I do see the RFC clock management API https://github.com/zephyrproject-rtos/zephyr/pull/72102 working for something like this when that comes

Reply:
Here we should be using `clocks = <&clock_controller ...>;`. I know that Zephyr doesn't have a proper clock control API making DT link smoother, but we should still write proper DT and add the necessary glue code in C in the meantime. For example, in the case of Nordic, the clocks are described like this, so we'd be forced to hack our DT to use `input-clock-frequency` when we already have a properly defined clock controller. In some cases, the frequency may not even be computable at compile time.

Reply:
this is now completed... 'resolve the conversation' if this is what you were looking for?